etc/docker/Dockerfile

```diff
@@ -0,0 +1,205 @@
+# Multi-stage Dockerfile
+# Stages:
+#   1) distro-deps   - Install OS-level toolchain and libraries (base for all other stages)
+#   2) source-deps   - Install source-built dependencies (LLVM 11) on top of distro-deps
+#   3) app-source-*  - Prepare application sources (from local context or remote repo)
+#   4) app-build-*   - Build the application from the chosen app-source-* stage
+#
+# Use `--target` at build time to select which final build stage to produce:
+#   docker build --target=app-build-local -t wire-sysio:local .
+#   docker build --target=app-build-repo  -t wire-sysio:repo  .
+
+# Stage 1: distro-deps
+# Base image with system packages required for building (compilers, cmake, Python, etc.)
+FROM ubuntu:24.04 AS distro-deps
+
+ENV DEBIAN_FRONTEND=noninteractive
+ENV PLATFORM_DIR=/wire
+ENV APP_DIR="${PLATFORM_DIR}/wire-sysio"
+ENV CDT_DIR="${PLATFORM_DIR}/wire-cdt"
+
+ENV BUILD_TYPE=Release
+ENV APP_BUILD_DIR="${APP_DIR}/build/${BUILD_TYPE}"
+ENV CDT_BUILD_DIR="${CDT_DIR}/build/${BUILD_TYPE}"
+
+ENV APP_DEV_BUILD_DIR="${APP_DIR}/build/debug-docker"
+ENV CDT_DEV_BUILD_DIR="${CDT_DIR}/build/debug-docker"
+
+ENV LLVM_11_DIR=/opt/llvm/llvm-11
+
+# Core package tools and repository helpers
+RUN apt-get update
+RUN apt-get install -y \
+        lsb-release \
+        wget \
+        software-properties-common
+
+# Enable Python 3.12 PPA and refresh indexes
+RUN add-apt-repository ppa:deadsnakes/ppa -y
+RUN apt-get update
+
+# Build toolchain and development libraries needed by the application
+RUN apt-get install -y \
+        gnupg \
+        build-essential \
+        ccache \
+        ninja-build \
+        cmake \
+        curl \
+        zip \
+        unzip \
+        tar \
+        git \
+        vim \
+        sudo \
+        doxygen \
+        golang \
+        llvm-18 \
+        clang-18 \
+        clang-tools-18 \
+        libboost-all-dev \
```

Comment:
This should not be needed as vcpkg is going to pull in a specific version.

Reply:
Probably not, but I'm not going to remove at the moment as everything is seemingly functional at the moment. @bearcubsvet once this lands please go through and attempt to remove those deps